app/src/main/java/org/wikipedia/page/leadimages/LeadImagesHandler.java

```diff
@@ -134,40 +147,60 @@ private void updateCallToAction() {
+                            }
+                            if (app.language().getAppLanguageCodes().size() >= MIN_LANGUAGES_TO_UNLOCK_TRANSLATION) {
+                                for (String lang : app.language().getAppLanguageCodes()) {
+                                    if (!imageCaptions.containsKey(lang)) {
+                                        imageEditType = ImageEditType.ADD_CAPTION_TRANSLATION;
+                                        captionTargetPageTitle = new PageTitle(imageTitle, new WikiSite(Service.COMMONS_URL, lang));
+                                        break;
+                                    }
+                                }
                             }
+                            return ImageTagsProvider.getImageTagsObservable(pair.second.query().firstPage().pageId(), getTitle().getWikiSite().languageCode());
                         }
+                )
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(imageTagsResult -> {
+                    if (imageEditType != ImageEditType.ADD_CAPTION && imageTagsResult != null && imageTagsResult.size() == 0) {
+                        imageEditType = ImageEditType.ADD_TAGS;
                     }
-                }, L::e));
+                    updateCta();
+                })
+        );
+    }
+
+    private void updateCta() {
+        switch (imageEditType) {
+            case ADD_TAGS:
+                pageHeaderView.setUpCallToAction(app.getResources().getString(R.string.suggested_edits_article_cta_image_tags));
+                break;
+            case ADD_CAPTION_TRANSLATION:
+                callToActionIsTranslation = true;
+                String currentCaption = imageCaptions.get(getTitle().getWikiSite().languageCode());
+                captionSourcePageTitle.setDescription(currentCaption);
+                callToActionSourceSummary = new PageSummaryForEdit(captionSourcePageTitle.getPrefixedText(), captionSourcePageTitle.getWikiSite().languageCode(), captionSourcePageTitle,
+                        captionSourcePageTitle.getDisplayText(), currentCaption, getLeadImageUrl());
+
+                callToActionTargetSummary = new PageSummaryForEdit(captionTargetPageTitle.getPrefixedText(), captionTargetPageTitle.getWikiSite().languageCode(), captionTargetPageTitle,
+                        captionTargetPageTitle.getDisplayText(), null, getLeadImageUrl());
+                pageHeaderView.setUpCallToAction(app.getResources().getString(R.string.suggested_edits_article_cta_image_caption_in_language, app.language().getAppLanguageLocalizedName(captionTargetPageTitle.getWikiSite().languageCode())));
+                break;
+            default:
+                pageHeaderView.setUpCallToAction(app.getResources().getString(R.string.suggested_edits_article_cta_image_caption));
```

Comment:
Can we move this line after the `callToActionSourceSummary` just like other ones?